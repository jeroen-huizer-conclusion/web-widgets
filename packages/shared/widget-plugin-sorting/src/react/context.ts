import { Context, createContext, useContext } from "react";
import { SortStoreHost } from "../stores/SortStoreHost";
import { SortInstruction } from "../types/store";

export interface SortAPI {
    version: 1;
    host: SortStoreHost;
    initSortOrder?: SortInstruction[];
}

const SORT_PATH = "com.mendix.widgets.web.sortable.sortContext";

export function getGlobalSortContext(): Context<SortAPI | null> {
    return ((window as any)[SORT_PATH] ??= createContext<SortAPI | null>(null));
}

export function useSortAPI(): SortAPI {
    const api = useContext(getGlobalSortContext());
    if (api === null) {
        throw new Error("SortAPI is not available. Widget is out of parent context.");
    }
    return api;
}
