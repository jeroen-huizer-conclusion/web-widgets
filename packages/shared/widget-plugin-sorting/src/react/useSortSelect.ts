import { useSetup } from "@mendix/widget-plugin-mobx-kit/react/useSetup";
import { SingleSortController } from "../controllers/SingleSortController";
import { BasicSortStore, SortDirection } from "../types/store";

interface HookProps {
    emptyOptionCaption?: string;
    store: BasicSortStore;
}

interface Option {
    caption: string;
    value: string;
}

interface ControlProps {
    value: string | null;
    options: Option[];
    direction: SortDirection;
    onSelect: (value: string) => void;
    onDirectionClick: () => void;
}

export function useSortSelect(props: HookProps, store: BasicSortStore): ControlProps {
    const ctrl = useSetup(() => new SingleSortController({ store }));

    return {
        value: ctrl.selected,
        options: ctrl.options,
        direction: ctrl.direction,
        onSelect: ctrl.select,
        onDirectionClick: ctrl.toggleDirection
    };
}
