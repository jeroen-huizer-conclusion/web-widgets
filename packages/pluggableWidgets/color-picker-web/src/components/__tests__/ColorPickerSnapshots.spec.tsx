import "@testing-library/jest-dom";
import { render, RenderResult } from "@testing-library/react";
import { createElement } from "react";
import { ColorPicker, ColorPickerProps } from "../ColorPicker";

jest.mock("react-color/lib", () => ({
    SketchPicker: "sketch-picker"
}));

describe("ColorPicker", () => {
    let colorPickerProps: ColorPickerProps;

    beforeEach(() => {
        colorPickerProps = {
            color: "#000000",
            disabled: false,
            defaultColors: [],
            format: "hex",
            mode: "popover",
            type: "sketch",
            onChange: jest.fn(),
            onColorChange: jest.fn(),
            id: "color-picker",
            name: "color picker"
        };
    });

    function renderColorPicker(configs: Partial<ColorPickerProps> = {}): RenderResult {
        return render(<ColorPicker {...colorPickerProps} {...configs} />);
    }

    describe("with a mode as", () => {
        it("popover or input renders with the structure", () => {
            const colorPickerComponent = renderColorPicker({ mode: "popover" });

            expect(colorPickerComponent.asFragment()).toMatchSnapshot();
        });

        it("inline renders with the structure", () => {
            const colorPickerComponent = renderColorPicker({ mode: "inline" });

            expect(colorPickerComponent.asFragment()).toMatchSnapshot();
        });
    });
});
