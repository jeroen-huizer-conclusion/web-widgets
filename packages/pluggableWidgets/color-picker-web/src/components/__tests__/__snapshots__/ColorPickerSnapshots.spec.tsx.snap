// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker with a mode as inline renders with the structure 1`] = `
<DocumentFragment>
  <div
    class="widget-color-picker widget-color-picker-picker"
  >
    <div
      class=""
    >
      <sketch-picker
        color="#000000"
        disablealpha="true"
        disabled="false"
        displaycolorpicker="false"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ColorPicker with a mode as popover or input renders with the structure 1`] = `
<DocumentFragment>
  <div
    class="widget-color-picker widget-color-picker-picker"
  >
    <button
      class="btn"
    >
      <div
        class="widget-color-picker-inner"
        style="background: rgb(0, 0, 0);"
      />
    </button>
  </div>
</DocumentFragment>
`;
